src/Framework/Framework/ViewModel/Validation/ValidationErrorFactory.cs

```diff
@@ -20,22 +21,25 @@ namespace DotVVM.Framework.ViewModel.Validation
 {
     public static class ValidationErrorFactory
     {
-        public static ViewModelValidationError AddModelError<T, TProp>(this T vm, Expression<Func<T, TProp>> expr, string message)
-            where T : class, IDotvvmViewModel =>
-            AddModelError(vm, expr, message, vm.Context);
+        public static ViewModelValidationError AddModelError<T>(this T vm, string message)
+            where T : class, IDotvvmViewModel
+            => vm.Context.AddModelError(message);
+
+        public static ViewModelValidationError AddModelError<T>(this T vm, string propertyPath, string message)
+            where T : class, IDotvvmViewModel
+            => vm.Context.AddModelError(propertyPath, message);
```

Comment:
And this too.

Reply:
I don't see why it would be helpful here. We do not work much with errors that have hand-written property paths. The only thing we could check is whether `vm` is reachable from root viewmodel.

But then again, I remember us having discussion that this API is only "useful" when user is doing some weird stuff with viewmodels in javascript. Due to this reason I don't feel like we want to do any checks here. What do you think?

Reply:
After internal discussion we decided that this overload will be removed.